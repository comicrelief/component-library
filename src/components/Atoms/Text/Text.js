import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/** Text component */
export const BaseText = styled.span`
  color: ${({ color, theme }) => (color ? theme.color(color) : 'inherit')};
  font-size: ${({ size, theme }) => theme.fontSize(size)};
  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'inherit')};
  ${({ weight }) => (weight ? `font-weight: ${weight}` : null)};
  ${({ height }) => (height ? `line-height: ${height}` : null)};
  font-family: ${({ family, theme }) =>
    family ? theme.fontFamilies(family) : 'inherit'};
`;

/** Text renders different elements based on the `tag` prop
 *  Weight is checked for existence to prevent overriding the tag's css
 */
const Text = ({ tag, children, height, weight, family, ...rest }) => (
  <BaseText {...rest} as={tag} height={height} weight={weight} family={family}>
    {children}
  </BaseText>
);

Text.defaultProps = {
  family: 'Montserrat',
  tag: 'span',
  weight: undefined,
  height: undefined,
  uppercase: false,
  size: 's',
  color: 'inherit'
};

Text.propTypes = {
  /** Font family */
  family: PropTypes.string,
  /** Font weight */
  weight: PropTypes.string,
  /** Line height */
  height: PropTypes.string,
  /** Sets text transform to uppercase. */
  uppercase: PropTypes.bool,
  /** Colors */
  color: PropTypes.string,
  /** Tag type */
  tag: PropTypes.string,
  /** Sizes */
  size: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string
  ]).isRequired
};

export default Text;
