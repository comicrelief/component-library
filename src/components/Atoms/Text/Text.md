```js
<div>
  <Text tag="h3" size="s">
    Heading 3
  </Text>
  <Text tag="p" colour="black">
    My paragraph
  </Text>
  <Text tag="h3" size="m">
    Heading 3 size 3
  </Text>
  <Text tag="p" size="xl" color="red">
    My paragraph small and yellow
  </Text>
</div>
```

Sport Relief

```js
  <div>
    <Text tag="h3" size="s">
      Heading 3
    </Text>
    <Text tag="p" colour="black">
      My paragraph
    </Text>
    <Text tag="h3" size="m">
      Heading 3 size 3
    </Text>
    <Text tag="p" size="xl" color="red">
      My paragraph small and yellow
    </Text>
  </div>
```
