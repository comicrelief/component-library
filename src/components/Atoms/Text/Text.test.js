import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import Text from './Text';

it('renders high heading correctly', () => {
  const tree = renderWithTheme(
    <Text tag="h3" height="5">
      Heading 3 high
    </Text>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders heavy heading correctly', () => {
  const tree = renderWithTheme(
    <Text tag="h3" weight="1000">
      Heading 3 heavy
    </Text>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders large heading correctly', () => {
  const tree = renderWithTheme(
    <Text tag="h3" size="l">
      Heading 3 large
    </Text>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders xl yellow paragraph correctly', () => {
  const tree = renderWithTheme(
    <Text tag="p" size="xl" color="yellow">
      My paragraph xl and yellow
    </Text>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders xl paragraph with Anton font correctly', () => {
  const tree = renderWithTheme(
    <Text tag="p" size="xl" family="Anton">
      My paragraph xl and font Anton
    </Text>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
