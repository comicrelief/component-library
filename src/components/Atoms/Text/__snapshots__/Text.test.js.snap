// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heavy heading correctly 1`] = `
.c0 {
  font-size: 1rem;
  text-transform: inherit;
  font-weight: 1000;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

<h3
  className="c0"
  color="inherit"
  size="s"
>
  Heading 3 heavy
</h3>
`;

exports[`renders high heading correctly 1`] = `
.c0 {
  font-size: 1rem;
  text-transform: inherit;
  line-height: 5;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

<h3
  className="c0"
  color="inherit"
  height="5"
  size="s"
>
  Heading 3 high
</h3>
`;

exports[`renders large heading correctly 1`] = `
.c0 {
  font-size: 1.44rem;
  text-transform: inherit;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

<h3
  className="c0"
  color="inherit"
  size="l"
>
  Heading 3 large
</h3>
`;

exports[`renders xl paragraph with Anton font correctly 1`] = `
.c0 {
  font-size: 1.725rem;
  text-transform: inherit;
  font-family: 'Anton',Impact,sans-serif;
}

<p
  className="c0"
  color="inherit"
  size="xl"
>
  My paragraph xl and font Anton
</p>
`;

exports[`renders xl yellow paragraph correctly 1`] = `
.c0 {
  color: #FFE400;
  font-size: 1.725rem;
  text-transform: inherit;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

<p
  className="c0"
  color="yellow"
  size="xl"
>
  My paragraph xl and yellow
</p>
`;
